src/main.rs

```diff
@@ -118,6 +118,19 @@ fn run() -> Result<()> {
                 .action(ArgAction::SetTrue)
                 .help("Use array increment for cluster size"),
         )
+        .arg(
+            Arg::new("impl_register_debug")
+                .long("impl_register_debug")
+                .action(ArgAction::SetTrue)
+                .help("implement Debug for readable registers"),
+        )
+        .arg(
```

Comment:
this also includes more than just registers now (blocks specifically), maybe rename it to

```
impl_debug
```